### PR TITLE
feat(plugins): add call for livestorm review command, liv-4506

### PR DESCRIPTION
**Pull request by Michaelvilleneuve**

### CODE REVIEW

File: src/cmds/review.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+module.exports = async function review() {
+  try {
+    const config = getLivestormPluginInformation('production')
+    const answers = await prompts(questions)
+    const zipUrl = await uploadZip(createZip())
+  
+    console.log('Sending review request...')
+  
+    const res = await fetch(`https://plugins.livestorm.co/api/v1/plugins/${config.name}/reviews`, {
```
Comment (jinevo) [inline src/cmds/review.js:73]: Je me demande si on ne devrait pas stocker l"url `https://plugins.livestorm.co/api/v1/plugins` en tant que variable d'environnement privée pour que ce soit plus simple à maintenir en cas de modificiation.

Comment (Michaelvilleneuve) [reply src/cmds/review.js:73]: Ahh c'est intéressant car ça permettrait de tester facilement sur un autre env

T'imagine un truc du genre : 
```
const BASE_URL =  process.env.LIVESTORM_DOMAIN || 'https://plugins.livestorm.co/'
```
j'aime bien car pour nous en staging on pourrait : 
```
livestorm review LIVESTORM_DOMAIN='https://xxxx'
```

**Review by jinevo — APPROVED** [review]

**Comment by jinevo** [comment]
I wonder if the CLI should be written in TypeScript to improve the robustness and to avoid silly errors, what do you think ?

File: src/cmds/review.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+const uploadFileOrDirectory = require('../helpers/uploadFileOrDirectory')
+const getLivestormPluginInformation = require('../helpers/getLivestormPluginInformation')
+const livestormDomain = require('../helpers/livestormDomain')
+
+const questions = [
+  {
+    type: 'multiselect',
+    name: 'purposes',
+    message: 'What does your plugin need a review for ?',
```
Comment (Pygocentrus) [inline src/cmds/review.js:16]: Wording: `What does your plugin need a review for?` (pas d'espace avant le ?)

File: src/cmds/review.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    choices: [
+      { title: 'I want to publish to the marketplace', value: 'marketplace' },
```
Comment (Pygocentrus) [inline src/cmds/review.js:18]: `I want to publish it in Livestorm\'s apps marketplace`

File: src/cmds/review.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      { title: 'I need the allow-same-origin flag', value: 'allow-same-origin' },
```
Comment (Pygocentrus) [inline src/cmds/review.js:19]: `I need to use the \'allow-same-origin\' flag on an iframe`

File: src/cmds/review.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      { title: 'I need to use a private API', value: 'private-api' }
+    ],
+    hint: 'Use arrows to browse and space bar to toggle',
+    instructions: false,
+    min: 1
+  },
+  {
+    type: 'text',
+    name: 'comment',
+    message: 'Anything in particular you want to communicate to the review team ?',
```
Comment (Pygocentrus) [inline src/cmds/review.js:29]: `Is there anything else you\'d like to communicate to the review team?`

File: src/cmds/review.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        zipUrl
+      }),
+      headers: {
+        'Content-Type': 'Application/JSON',
+        'Authorization': config.apiToken || config.apiKey,
+      }
+    })
+  
+    if (res.status === 201) {
+      console.log(`Done ! You review request has been created, we'll get back to you by email at ${answers.email}.`)
```
Comment (Pygocentrus) [inline src/cmds/review.js:93]: `Done! Your review has been sent, we'll get back to you shortly at ${answers.email}.`

**Review by Pygocentrus — COMMENTED** [review]
@Michaelvilleneuve très cool, juste quelques typos/wordings suggérés ✌️

File: src/cmds/review.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+function createZip() {
+  console.log('Deleting node_modules folder to reduce archive size...')
+  rimraf.sync('./node_modules')
+  console.log('Creating zip...')
+
+  const name = `tmp-${crypto.randomBytes(15).toString('hex')}.zip`
+  const zip = new Zip()
+  zip.addLocalFolder('./')
+  zip.deleteFile('__tests__/')
```
Comment (jinevo) [inline src/cmds/review.js:50]: Why delete after creating of the zip and not before like the `node_modules` ?

File: src/cmds/review.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+async function uploadZip(zipFile) {
+  console.log('Uploading zip...')
+  const url = await uploadFileOrDirectory(zipFile, false)
+  rimraf.sync(`./${zipFile}`)
+  return url
+}
+
+module.exports = async function review() {
+  try {
+    const config = getLivestormPluginInformation('production')
```
Comment (jinevo) [inline src/cmds/review.js:70]: `production` could be a key of an enum (or object) ?

**Pull request merged**